src/test/java/org/gaul/s3proxy/EncryptedBlobStoreLiveTest.java

```diff
@@ -0,0 +1,287 @@
+        String eTagOf2 = this.getApi()
+            .uploadPart(containerName, blobName, 2, uploadId, part2);
+        String eTagOf3 = this.getApi()
+            .uploadPart(containerName, blobName, 3, uploadId, part3);
+
+        this.getApi().completeMultipartUpload(containerName, blobName, uploadId,
+            ImmutableMap.of(1, eTagOf1, 2, eTagOf2, 3, eTagOf3));
+        S3Object object = this.getApi().getObject(containerName, blobName);
+
+        try (InputStream actual = object.getPayload().openStream();
+             InputStream expected = byteSource.openStream()) {
+            assertThat(actual).hasContentEqualTo(expected);
+        }
+
+        // get a 5mb slice that overlap parts
+        long partialStart = 5 * 1024 * 1024;
+        ByteSource partialContent =
+            byteSource.slice(partialStart, partialStart);
+
+        GetOptions options = new GetOptions();
+        options.range(partialStart, (partialStart * 2) - 1);
+        object = this.getApi().getObject(containerName, blobName, options);
+
+        try (InputStream actual = object.getPayload().openStream();
+             InputStream expected = partialContent.openStream()) {
+            assertThat(actual).hasContentEqualTo(expected);
+        }
+    }
+
+    @Override
+    public void testMultipartSynchronously() {
+        throw new SkipException("list multipart synchronously not supported");
+    }
+
+    @Override
+    @Test
+    public void testUpdateObjectACL() throws InterruptedException,
+        ExecutionException, TimeoutException, IOException {
+        try {
+            super.testUpdateObjectACL();
+            Fail.failBecauseExceptionWasNotThrown(AWSResponseException.class);
+        } catch (AWSResponseException are) {
+            assertThat(are.getError().getCode()).isEqualTo("NotImplemented");
+            throw new SkipException("XML ACLs not supported", are);
+        }
+    }
+
+    @Override
+    @Test
+    public void testPublicWriteOnObject() throws InterruptedException,
+        ExecutionException, TimeoutException, IOException {
+        try {
+            super.testPublicWriteOnObject();
+            Fail.failBecauseExceptionWasNotThrown(AWSResponseException.class);
+        } catch (AWSResponseException are) {
+            assertThat(are.getError().getCode()).isEqualTo("NotImplemented");
+            throw new SkipException("public-read-write-acl not supported", are);
+        }
+    }
+
+    @Override
+    public void testCopyCannedAccessPolicyPublic() {
+        throw new SkipException("blob access control not supported");
+    }
+
+    @Override
+    public void testPutCannedAccessPolicyPublic() {
+        throw new SkipException("blob access control not supported");
+    }
+
+    @Override
+    public void testUpdateObjectCannedACL() {
+        throw new SkipException("blob access control not supported");
+    }
+
+    @Override
+    public void testCopyIfMatch() {
+        throw new SkipException("copy if match not supported");
```

Comment:
If you get rid of the ETag munging will this test case pass?

Reply:
reverted my changes